src/agreement.rs

```diff
@@ -216,12 +210,13 @@ impl<NodeUid: Clone + Eq + Hash> Agreement<NodeUid> {
     /// can, however, expect every good node to send an AUX value that will
     /// eventually end up in our bin_values.
     fn count_aux(&self) -> (usize, BTreeSet<bool>) {
-        let vals: BTreeSet<bool> = self.received_aux
-            .values()
-            .filter(|values| values.is_subset(&self.bin_values))
-            .fold(BTreeSet::new(), |vals, values| {
-                vals.union(values).cloned().collect()
-            });
+        let mut vals: BTreeSet<bool> = BTreeSet::new();
+
+        for b in self.received_aux.values() {
+            if self.bin_values.contains(b) {
+                vals.insert(b.clone());
+            }
+        }
```

Comment:
But now `vals.len()` will always be `<= 2` (number of values instead of number of entries). Maybe it should be:

```rust
let aux_count = self.received_aux.values()
    .filter(|b| self.bin_values.contains(b))
    .count();
(aux_count, vals)
```

Reply:
How do you define `vals`?

Reply:
The same way you already do.

Reply:
If so then `aux_count == vals.len()`. So, `aux_count` is redundant.